Makefile

```diff
@@ -1,55 +1,50 @@
-VIRTUALENV = virtualenv
 VENV := $(shell echo $${VIRTUAL_ENV-$$PWD/.venv})
 PYTHON = $(VENV)/bin/python
-DEV_STAMP = $(VENV)/.dev_env_installed.stamp
 INSTALL_STAMP = $(VENV)/.install.stamp
-TEMPDIR := $(shell mktemp -d)
 
 .IGNORE: clean
-.PHONY: all install virtualenv tests install-dev tests-once
+.PHONY: all install virtualenv tests tests-once
 
 OBJECTS = .venv .coverage
 
 all: install
-install: $(INSTALL_STAMP)
-$(INSTALL_STAMP): $(PYTHON) requirements.txt setup.py
+
+install: $(INSTALL_STAMP) pyproject.toml requirements.txt
+$(INSTALL_STAMP): $(PYTHON) pyproject.toml requirements.txt
 	$(VENV)/bin/pip install -r requirements.txt
-	$(PYTHON) setup.py develop
+	$(VENV)/bin/pip install ".[dev]"
 	touch $(INSTALL_STAMP)
 
-install-dev: $(INSTALL_STAMP) $(DEV_STAMP)
-$(DEV_STAMP): $(PYTHON) dev-requirements.txt
-	$(VENV)/bin/pip install -r dev-requirements.txt
-	touch $(DEV_STAMP)
-
 virtualenv: $(PYTHON)
 $(PYTHON):
-	virtualenv $(VENV)
+	python -m venv $(VENV)
+
+lint: install
+	$(VENV)/bin/ruff check kinto_emailer *.py
+	$(VENV)/bin/ruff format --check kinto_emailer *.py
+
+format: install
+	$(VENV)/bin/ruff check --fix kinto_emailer *.py
+	$(VENV)/bin/ruff format kinto_emailer *.py
 
 build-requirements:
-	$(VIRTUALENV) $(TEMPDIR)
-	$(TEMPDIR)/bin/pip install -U pip
-	$(TEMPDIR)/bin/pip install -Ue .
-	$(TEMPDIR)/bin/pip freeze | grep -v -- '^-e' > requirements.txt
+	pip-compile --generate-hashes requirements.in > requirements.txt
```

Comment:
I saw this in the [pip-tools docs](https://pip-tools.readthedocs.io/en/latest/#requirements-from-pyproject-toml):

```
pip-compile -o requirements.txt pyproject.toml
```

would it be worth it for us to do something similar? I'm not sure what advantage it has over the way we do it here, though.

Reply:
Also, it looks like we can specify some of these CLI args as config options in pyproject.toml, if we want https://pip-tools.readthedocs.io/en/latest/#configuration